### PR TITLE
Update the documentation on upgrading Marathon

### PR DESCRIPTION
**Pull request by ichernetsky**
JIRA issues: MARATHON-2447

### CODE REVIEW

File: docs/docs/upgrade/index.md [hunk 1]
```diff
@@ -24,6 +24,113 @@ Upgrading to a newer version of Marathon should be executed in the following ord
...
+
+For further details please refer to the changelog.
+
+# Upgrading to 1.5
+
+Release Notes: https://github.com/mesosphere/marathon/releases/tag/v1.5.0
+
+The recommended Mesos version is 1.3.0 or later. Upgrade to Marathon 1.5.x can be performed only from 1.4.x.
+
+## Breaking Changes
```
Comment (jeschkies) [inline docs/docs/upgrade/index.md]: Hm, shouldn't this be part of the release notes? It seems to be an overhead of managing these in addition to the change notes. What do you think?

File: changelog.md [hunk 2]
```diff
@@ -146,7 +146,7 @@ This plugin allows to reject offers. Possible use-cases are:
 
 ## Changes from 1.4.x to 1.5.0
 
-### Recommended Mesos version is 1.3.0
+Recommended Mesos version is 1.3.0. Upgrade to Marathon 1.5.x can be performed only from 1.4.x.
```
Comment (timcharper) [inline changelog.md]: > Recommended Mesos version is 1.3.0 or higher/later?

File: docs/docs/upgrade/index.md (hunk 1 above)
Comment (timcharper) [reply docs/docs/upgrade/index.md]: I'm inclined to agree. This stuff (breaking changes) should just be in the release notes.

**Comment by ichernetsky** [comment]
@timcharper @jeschkies I addressed your concerns. Please take a look.

**Comment by ichernetsky** [comment]
@timcharper, could you please review it one more time?

**Pull request merged**